exporter/stackdriverexporter/stackdriver.go

```diff
@@ -116,14 +117,24 @@ func newStackdriverMetricsExporter(cfg *Config) (component.MetricsExporter, erro
 		// Set DefaultMonitoringLabels to an empty map to avoid getting the "opencensus_task" label
 		DefaultMonitoringLabels: &stackdriver.Labels{},
 	}
+
+	userAgent := strings.ReplaceAll(cfg.UserAgent, "{{version}}", version)
 	if cfg.Endpoint != "" {
-		copts, err := generateClientOptions(cfg)
+		// WithGRPCConn option takes precedent over all other supplied options so need to provide user agent here as well
+		dialOpts := []grpc.DialOption{}
+		if userAgent != "" {
```

Comment:
Given that OpenCensus exporter sets user agent to `"opencensus-go %s; stackdriver-exporter %s"` by default, we may want to override it with `"opentelemetry-collector-contrib {{version}}"` by default (i.e. even if it's not configured)?